cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -628,41 +733,86 @@ class ArrowOutputStream : public liborc::OutputStream {
   int64_t length_;
 };
 
+Result<liborc::WriterOptions> MakeOrcWriterOptions(
+    arrow::adapters::orc::WriteOptions options) {
+  liborc::WriterOptions orc_options_;
+  orc_options_.setFileVersion(
+      liborc::FileVersion(static_cast<uint32_t>(options.file_version.major()),
+                          static_cast<uint32_t>(options.file_version.minor())));
+  orc_options_.setStripeSize(static_cast<uint64_t>(options.stripe_size));
+  orc_options_.setCompressionBlockSize(
+      static_cast<uint64_t>(options.compression_block_size));
+  orc_options_.setCompressionStrategy(static_cast<liborc::CompressionStrategy>(
+      static_cast<int8_t>(options.compression_strategy)));
+  orc_options_.setRowIndexStride(static_cast<uint64_t>(options.row_index_stride));
+  orc_options_.setPaddingTolerance(options.padding_tolerance);
+  orc_options_.setDictionaryKeySizeThreshold(options.dictionary_key_size_threshold);
+  orc_options_.setPaddingTolerance(options.padding_tolerance);
+  std::set<uint64_t> orc_bloom_filter_columns_;
+  std::for_each(options.bloom_filter_columns.begin(), options.bloom_filter_columns.end(),
+                [&orc_bloom_filter_columns_](const int64_t col) {
+                  orc_bloom_filter_columns_.insert(static_cast<uint64_t>(col));
+                });
+  orc_options_.setColumnsUseBloomFilter(orc_bloom_filter_columns_);
```

Comment:
```suggestion
  orc_options_.setColumnsUseBloomFilter(std::move(orc_bloom_filter_columns_));
```

Reply:
Fixed

Reply:
Sorry..currently we don’t have a shared pointer to a `std::vector<int32_t>`

Reply:
Hmm, so what? `orc_bloom_filter_columns_` is a local variable and it isn't used anymore after this, it's not a problem to move from it.